### ISSUE TITLE
ConcurrentModificationException

### ISSUE THREAD

**Issue by egkz** [gotev/android-cookie-store#4]
happens on this line  `preferences.edit().putString(index.toString(), gson.toJson(cookies)).apply()`

```
E/AndroidRuntime: FATAL EXCEPTION: OkHttp Dispatcher
    Process: com.light.finder, PID: 28247
    java.util.ConcurrentModificationException
        at java.util.ArrayList$Itr.next(ArrayList.java:860)
        at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.write(CollectionTypeAdapterFactory.java:96)
        at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.write(CollectionTypeAdapterFactory.java:61)
        at com.google.gson.Gson.toJson(Gson.java:704)
        at com.google.gson.Gson.toJson(Gson.java:683)
        at com.google.gson.Gson.toJson(Gson.java:638)
        at com.google.gson.Gson.toJson(Gson.java:618)
        at com.light.finder.data.source.remote.SharedPreferencesCookieStore.add(SharedPreferencesCookieStore.kt:51)
        at java.net.CookieManager.put(CookieManager.java:360)
        at com.light.finder.data.source.remote.WebKitSyncCookieManager.put(WebKitSyncCookieManager.kt:24)
        at okhttp3.JavaNetCookieJar.saveFromResponse(JavaNetCookieJar.kt:38)
        at okhttp3.internal.http.HttpHeaders.receiveHeaders(HttpHeaders.kt:207)
        at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:86)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:112)
        at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:71)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:112)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:87)
        at okhttp3.logging.HttpLoggingInterceptor.intercept(HttpLoggingInterceptor.java:213)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:112)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:87)
        at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.kt:184)
        at okhttp3.RealCall$AsyncCall.run(RealCall.kt:136)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:919)
```

**Comment by gotev** [comment]
please provide an isolated case in which this happens using only library classes. A test is better. `com.light.finder.data.source.remote.SharedPreferencesCookieStore` and `com.light.finder.data.source.remote.WebKitSyncCookieManager` are part of your own implementation.

**Comment by egkz** [comment]
2020-04-17 11:01:57.995 7627-8071/com.light.finder E/AndroidRuntime: FATAL EXCEPTION: OkHttp Dispatcher
    Process: com.light.finder, PID: 7627
    java.util.ConcurrentModificationException
        at java.util.ArrayList$Itr.next(ArrayList.java:860)
        at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.write(CollectionTypeAdapterFactory.java:96)
        at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.write(CollectionTypeAdapterFactory.java:61)
        at com.google.gson.Gson.toJson(Gson.java:704)
        at com.google.gson.Gson.toJson(Gson.java:683)
        at com.google.gson.Gson.toJson(Gson.java:638)
        at com.google.gson.Gson.toJson(Gson.java:618)
        at net.gotev.cookiestore.SharedPreferencesCookieStore.add(SharedPreferencesCookieStore.kt:51)
        at java.net.CookieManager.put(CookieManager.java:360)
        at net.gotev.cookiestore.WebKitSyncCookieManager.put(WebKitSyncCookieManager.kt:22)
        at okhttp3.JavaNetCookieJar.saveFromResponse(JavaNetCookieJar.kt:38)
        at okhttp3.internal.http.HttpHeaders.receiveHeaders(HttpHeaders.kt:207)
        at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:86)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:112)
        at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:71)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:112)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:87)
        at okhttp3.logging.HttpLoggingInterceptor.intercept(HttpLoggingInterceptor.java:213)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:112)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:87)
        at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.kt:184)
        at okhttp3.RealCall$AsyncCall.run(RealCall.kt:136)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:919)

**Comment by egkz** [comment]
might be related https://github.com/google/gson/issues/1159

**Comment by gotev** [comment]
Yeah it may be the case. Added synchronization blocks when reading/writing cookies and released `1.2.5`. Try it and let me know.
